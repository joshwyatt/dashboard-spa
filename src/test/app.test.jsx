import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';
/* eslint-disable import/extensions */
import { App, mapStateToProps, mapDispatchToProps } from '../App.jsx';

const state = {
  widgets: {
    ids: ['github'],
    showSidebar: true,
    grid: {
      layout: [{ i: 'slack', x: 0, y: 8, w: 6, h: 8, static: true }],
    },
  },
};
/* eslint-disable react/jsx-boolean-value */
describe('App component', () => {
  it('should render the App component with props as specified ', () => {
    const showAddWidgetModal = jest.fn();
    const showDashboardSidebar = jest.fn();
    const hideDashboardSidebar = jest.fn();
    const component = shallow(
      <App
        showAddWidgetModal={showAddWidgetModal}
        showDashboardSidebar={showDashboardSidebar}
        hideDashboardSidebar={hideDashboardSidebar}
        ids={['github']}
        showSidebar={true}
        layout={[{ i: 'slack', x: 0, y: 8, w: 6, h: 8, static: true }]}
      />,
    );
    expect(toJson(component)).toMatchSnapshot();
  });

  it('maps given state to props ', () => {
    const expected = {
      ids: ['github'],
      showSidebar: true,
      layout: [{ i: 'slack', x: 0, y: 8, w: 6, h: 8, static: true }],
    };
    expect(mapStateToProps(state)).toEqual(expected);
  });

  it('maps component dispatches to props ', () => {
    const dispatch = jest.fn();
    expect(mapDispatchToProps(dispatch)).toHaveProperty('showAddWidgetModal');
    expect(mapDispatchToProps(dispatch)).toHaveProperty('showDashboardSidebar');
    expect(mapDispatchToProps(dispatch)).toHaveProperty('hideDashboardSidebar');
  });

  it('Should have className page-container', () => {
    const showAddWidgetModal = jest.fn();
    const showDashboardSidebar = jest.fn();
    const hideDashboardSidebar = jest.fn();
    const component = shallow(
      <App
        showAddWidgetModal={showAddWidgetModal}
        showDashboardSidebar={showDashboardSidebar}
        hideDashboardSidebar={hideDashboardSidebar}
        ids={['github']}
        showSidebar={true}
        layout={[{ i: 'slack', x: 0, y: 8, w: 6, h: 8, static: true }]}
      />,
    );
    expect(component.find('.page-container').exists()).toBe(true);
  });
});
