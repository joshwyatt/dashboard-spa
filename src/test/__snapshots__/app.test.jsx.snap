// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component should render the App component with props as specified  1`] = `
<div
  className="page-container"
>
  <div
    className="grid-container"
  >
    <SidebarPushable
      as={[Function]}
    >
      <Sidebar
        animation="overlay"
        as={[Function]}
        direction="right"
        icon="labeled"
        inverted={true}
        vertical={true}
        visible={true}
        width="thin"
      >
        <MenuItem
          name="Add_Widget"
          onClick={[Function]}
        >
          <Icon
            as="i"
            name="add circle"
          />
          Add Widget
        </MenuItem>
        <MenuItem
          disabled={true}
          name="Settings"
        >
          <Icon
            as="i"
            name="cogs"
          />
          Settings
        </MenuItem>
      </Sidebar>
      <SidebarPusher>
        <Segment
          basic={true}
        >
          <_class
            className="layout"
            cols={12}
            layout={
              Array [
                Object {
                  "h": 8,
                  "i": "slack",
                  "static": true,
                  "w": 6,
                  "x": 0,
                  "y": 8,
                },
              ]
            }
            measureBeforeMount={false}
            rowHeight={30}
            verticalCompact={false}
            width={1200}
          >
            <div>
              <Connect(App)
                widgetId="github"
              />
            </div>
          </_class>
        </Segment>
      </SidebarPusher>
    </SidebarPushable>
  </div>
  <div
    className="sideStrip"
    onClick={[Function]}
    role="link"
    tabIndex="-1"
  >
    <Icon
      as="i"
      name="chevron right"
    />
  </div>
  <Connect(Component) />
</div>
`;
